src/AspNet.Security.OAuth.Bilibili/BilibiliAuthenticationHandler.cs

```diff
@@ -90,17 +90,16 @@ private static string ComputeHmacSHA256(string key, string data)
         return Convert.ToHexStringLower(hash);
     }
 
-    private static string BuildSignatureString(HttpRequestMessage request, string appSecret)
+    private static string BuildSignatureString(SortedList<string, string> xBiliHeaders, string appSecret)
     {
-        var headers = request.Headers
-            .Where(h => h.Key.StartsWith("x-bili-", StringComparison.OrdinalIgnoreCase))
-            .OrderBy(h => h.Key)
-            .Select(h => $"{h.Key}:{string.Join(",", h.Value)}")
-            .ToList();
-
-        var signature = string.Join('\n', headers);
+        StringBuilder sb = new(256); // 256 is an estimated size for the plain text
+        foreach (var xbh in xBiliHeaders)
+        {
+            sb.Append(xbh.Key).Append(':').Append(xbh.Value).Append('\n');
```

Comment:
Style preference for readability:

```suggestion
            builder.Append(xbh.Key)
                   .Append(':')
                   .Append(xbh.Value)
                   .Append('\n');
```